subsys/net/lib/tls_credentials/tls_credentials_digest_raw.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+/* This file provides an (internal-use-only) credential digest function that backends storing
+ * raw credentials can use.
+ */
+
+#include <string.h>
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include "tls_internal.h"
+#include "tls_credentials_digest_raw.h"
+
+/* Grab mbedTLS headers if they are available so that we can check whether SHA256 is supported */
+
+#if defined(CONFIG_MBEDTLS)
+#if !defined(CONFIG_MBEDTLS_CFG_FILE)
+#include "mbedtls/config.h"
+#else
+#include CONFIG_MBEDTLS_CFG_FILE
+#endif /* CONFIG_MBEDTLS_CFG_FILE */
+#endif /* CONFIG_MBEDTLS */
+
+#if defined(CONFIG_TINYCRYPT_SHA256) && defined(CONFIG_BASE64)
+
+#include <tinycrypt/sha256.h>
+#include <zephyr/sys/base64.h>
+
+struct tc_sha256_state_struct sha_state;
+uint8_t digest_buf[TC_SHA256_DIGEST_SIZE];
```

Comment:
Perhaps put these to stack (inside function) in order to allow use from multiple threads. Or if that is not needed, then at least have these as static.

Reply:
Great point, good catch